src/dialects/abstract/data-types.ts

```diff
@@ -0,0 +1,1826 @@
+    if (!options?.raw && value) {
+      return moment(value).format('YYYY-MM-DD');
+    }
+
+    return value;
+  }
+
+  protected _isChanged(value: AcceptedDate, originalValue: AcceptedDate) {
+    if (originalValue && Boolean(value) && originalValue === value) {
+      return false;
+    }
+
+    // not changed when set to same empty value
+    if (!originalValue && !value && originalValue === value) {
+      return false;
+    }
+
+    return true;
+  }
+}
+
+/**
+ * A date only column (no timestamp)
+ */
+export type DATEONLY = _DATEONLY;
+export const DATEONLY = classToInvokable(_DATEONLY);
+
+class _HSTORE extends ABSTRACT<Record<string, unknown>> {
+  public static readonly key = 'HSTORE';
+
+  public validate(value: Record<string, unknown>) {
+    const proto = Object.getPrototypeOf(value);
+
+    if (proto && proto !== Object.prototype) {
+      throw new ValidationError(
+        util.format('%j is not a valid hstore', value),
+        [],
+      );
+    }
+
+    return true;
+  }
+}
+
+/**
+ * A key / value store column. Only available in Postgres.
+ */
+export type HSTORE = _HSTORE;
+export const HSTORE = classToInvokable(_HSTORE);
+
+class _JSONTYPE extends ABSTRACT<any> {
+  public static readonly key: string = 'JSON';
+
+  validate() {
+    return true;
+  }
+
+  stringify(value: any) {
+    return JSON.stringify(value);
+  }
+}
+
+/**
+ * A JSON string column. Available in MySQL, Postgres and SQLite
+ */
+type JSONTYPE = _JSONTYPE;
+const JSONTYPE = classToInvokable(_JSONTYPE);
+
+export { JSONTYPE as JSON };
+
+class _JSONB extends JSONTYPE {
+  public static readonly key = 'JSONB';
+}
+
+/**
+ * A binary storage JSON column. Only available in Postgres.
+ */
+export type JSONB = _JSONB;
+export const JSONB = classToInvokable(_JSONB);
+
+class _NOW extends ABSTRACT<never> {
+  public static readonly key = 'NOW';
+}
+
+/**
+ * A default value of the current timestamp.  Not a valid type.
+ */
+export type NOW = _NOW;
+export const NOW = classToInvokable(_NOW);
+
+type AcceptedBlob = Buffer | string;
+
+enum BlobLength {
+  TINY = 'tiny',
+  MEDIUM = 'medium',
+  LONG = 'long',
+}
+
+interface BlobOptions {
+  length?: BlobLength;
+}
+
+class _BLOB extends ABSTRACT<AcceptedBlob, Buffer> {
+  public static readonly key = 'BLOB';
+  public static escape: typeof ABSTRACT['escape'] = false;
+  protected options: BlobOptions;
+  protected _length?: string;
+
+  /**
+   * @param [length=''] could be tiny, medium, long.
+   */
+  constructor(length?: BlobLength | BlobOptions) {
+    super();
+    const options: BlobOptions
+      = typeof length === 'object' ? length : { length };
+
+    this.options = options;
+    this._length = options.length?.toLowerCase();
+  }
+
+  toSql() {
+    switch (this._length) {
+      case BlobLength.TINY:
+        return 'TINY_BLOB';
+      case BlobLength.MEDIUM:
+        return 'MEDIUM_BLOB';
+      case BlobLength.LONG:
+        return 'LONG_BLOB';
```

Comment:
I don't think any dialect has `_` in their blob name https://new.sequelizejs.com/docs/v7/other-topics/other-data-types#blobs